### PR TITLE
[#IR-127] : Added support for sprint tasks and update all associated …

### PR DESCRIPTION
**Pull request by anmolghosh**
…queries and APIs

### CODE REVIEW

File: apps/retrospective/models/sprint_member_task.go [hunk 1]
```diff
@@ -54,38 +56,30 @@ type SprintMemberTask struct {
...
-
-	if sprintMemberTask.SprintMember.ID == 0 {
-		if err = db.Where("id = ?", sprintMemberTask.SprintMemberID).Find(&sprintMember).Error; err != nil {
-			return err
-		}
-	} else {
-		sprintMember = sprintMemberTask.SprintMember
-	}
-
+	sprintTaskFilter := db.Model(&SprintTask{}).Where("id = ?", sprintMemberTask.SprintTaskID).
```
Comment (himanshu-garg-165) [inline apps/retrospective/models/sprint_member_task.go]: Check for sprintTaskID from both the `SprintMemberTask.SprintTaskID` field and `SprintMemberTask.SprintTask.ID` to handle the case of SprintMemberTask create

File: apps/retrospective/services/sprint_task.go [hunk 2]
```diff
@@ -100,10 +116,10 @@ func (service TaskService) MarkDone(
 		return nil, http.StatusInternalServerError, errors.New("failed to mark the task as done")
 	}
 
+	query := db.Model(&retroModels.SprintTask{}).Where("id = ?", sprintTaskID).Select("id").QueryExpr()
 	err = db.Model(&retroModels.Task{}).
-		Where("tasks.id = ?", taskID).
-		Where("done_at is NULL").
-		Update("done_at", *sprint.EndDate).
+		Where("id = (?)", query).
```
Comment (himanshu-garg-165) [inline apps/retrospective/services/sprint_task.go:125]: the query variable would return the SprintTask ID not Task ID, so the update query will fail

File: apps/retrospective/services/sprint_task.go [hunk 3]
```diff
@@ -114,18 +130,18 @@ func (service TaskService) MarkDone(
...
+	sprintTaskID string,
 	retroID string,
-	sprintID string) (task *retroSerializers.Task, status int, err error) {
+	sprintID string) (task *retroSerializers.SprintTask, status int, err error) {
 	db := service.DB
+	query := db.Model(&retroModels.SprintTask{}).Where("id = ?", sprintTaskID).Select("id").QueryExpr()
 	err = db.Model(&retroModels.Task{}).
-		Where("tasks.id = ?", taskID).
-		Where("done_at is not NULL").
+		Where("id = (?)", query).
```
Comment (himanshu-garg-165) [inline apps/retrospective/services/sprint_task.go]: same as above

File: apps/retrospective/services/sprint_task.go [hunk 4]
```diff
@@ -54,17 +62,25 @@ func (service TaskService) List(
...
-	sprintID string) (task *retroSerializers.Task, status int, err error) {
+	sprintID string) (task *retroSerializers.SprintTask, status int, err error) {
 	db := service.DB
-	var tasks []retroSerializers.Task
+	var tasks []retroSerializers.SprintTask
 
-	dbs := service.tasksForActiveAndCurrentSprint(retroID, sprintID).
-		Where("tasks.id = ?", taskID).
+	dbs := service.tasksForActiveAndCurrentSprint(retroID, sprintID, &sprintTaskID).
+		Where("tasks.id = ?", sprintTaskID).
```
Comment (himanshu-garg-165) [inline apps/retrospective/services/sprint_task.go]: This should be `sprint_tasks.id`

**Review by himanshu-garg-165 — APPROVED** [review]

File: apps/retrospective/services/sprint_task.go (hunk 2 above)
Comment (anmolghosh) [reply apps/retrospective/services/sprint_task.go:125]: this comment was meant for line number above line. fixed

File: apps/retrospective/models/sprint_member_task.go (hunk 1 above)
Comment (anmolghosh) [reply apps/retrospective/models/sprint_member_task.go]: done

File: apps/retrospective/services/sprint_task.go (hunk 3 above)
Comment (anmolghosh) [reply apps/retrospective/services/sprint_task.go]: done

File: apps/retrospective/services/sprint_task.go (hunk 4 above)
Comment (anmolghosh) [reply apps/retrospective/services/sprint_task.go]: done

**Pull request merged**